### ISSUE TITLE
The meaning of the output metrics

### ISSUE THREAD

**Issue by BotaoPeng98** [roboflow/rf-detr#93]
Hope this message finds you well. I want to thank all of your team for releasing such a fabulous work and providing an easy-in-use and extensive toolkit of Detr.

When I was training a model, I didn't fully understand the meaning of the output metrics. 

```
Averaged stats: class_error: 0.00  loss: 5.4065 (5.4163)  loss_ce: 0.6641 (0.6705) 
loss_bbox: 0.0438 (0.0642)  loss_giou: 0.5722 (0.5855)  loss_ce_0: 0.6914 (0.6803)  
loss_bbox_0: 0.0451 (0.0633)  loss_giou_0: 0.5808 (0.5951)  loss_ce_1: 0.6523 (0.6669)  
loss_bbox_1: 0.0461 (0.0649)  loss_giou_1: 0.5597 (0.5916)  loss_ce_enc: 0.6914 (0.7007)  
loss_bbox_enc: 0.0626 (0.0756)  loss_giou_enc: 0.6624 (0.6578)  loss_ce_unscaled: 0.6641 (0.6705)  
class_error_unscaled: 0.0000 (0.3333)  loss_bbox_unscaled: 0.0088 (0.0128)  loss_giou_unscaled: 0.2861 (0.2928)  
cardinality_error_unscaled: 183.2500 (185.9767)  loss_ce_0_unscaled: 0.6914 (0.6803)  loss_bbox_0_unscaled: 0.0090 (0.0127)  
loss_giou_0_unscaled: 0.2904 (0.2976)  cardinality_error_0_unscaled: 207.7500 (207.0767)  loss_ce_1_unscaled: 0.6523 (0.6669)  
loss_bbox_1_unscaled: 0.0092 (0.0130)  loss_giou_1_unscaled: 0.2798 (0.2958)  cardinality_error_1_unscaled: 250.7500 (245.7700)  
loss_ce_enc_unscaled: 0.6914 (0.7007)  loss_bbox_enc_unscaled: 0.0125 (0.0151)  loss_giou_enc_unscaled: 0.3312 (0.3289)  
cardinality_error_enc_unscaled: 295.7500 (294.1433)
```

Is there any documentation for them? Or could you explain for us? 

BTW, the training process's observability is not very good. I appreciate future work to successfully improve them. 

BR,
Noah

**Comment by SkalskiP** [comment]
@probicheaux / @isaacrob-roboflow any chance you can help me out here?

**Comment by mkrupczak3** [comment]
I recommend following the instructions to set up Tensorboard. See also #78 for corrected training script for current version of rf-detr package

**Comment by isaacrob-roboflow** [comment]
@BotaoPeng98 those numbers are the values of the many loss functions used to train RF-DETR. the best way to understand them is to read the associated paper, but since we haven't released the paper yet I can give a very brief summary and link you to the relevant literature. 

we build off [LW-DETR](https://arxiv.org/abs/2406.03459) and use their loss functions. in tern, LW-DETR builds off [Group DETR](https://arxiv.org/pdf/2207.13085)

DETRs train with a bipartite matching loss which is worth reading directly, can be found in the original [DETR](https://arxiv.org/abs/2005.12872) paper. GIoU is a generalized IoU loss for bounding box localization, bbox loss is an L1 bounding box regression loss, and ce is an area-aware sigmoidal focal loss. class error is a percentage accuracy of classification for box classification and cardinality error is not worth paying attention to.

the rest of the meaning of the loss functions are most directly understood by ensuring familiarity with the code. when we train, we mainly monitor loss_ce_unscaled, class_error_unscaled, loss_bbox_unscaled, and loss_ce_unscaled, as well as their test and ema variants 

does that answer your question?

**Comment by BotaoPeng98** [comment]
> [@BotaoPeng98](https://github.com/BotaoPeng98) those numbers are the values of the many loss functions used to train RF-DETR. the best way to understand them is to read the associated paper, but since we haven't released the paper yet I can give a very brief summary and link you to the relevant literature.
> 
> we build off [LW-DETR](https://arxiv.org/abs/2406.03459) and use their loss functions. in tern, LW-DETR builds off [Group DETR](https://arxiv.org/pdf/2207.13085)
> 
> DETRs train with a bipartite matching loss which is worth reading directly, can be found in the original [DETR](https://arxiv.org/abs/2005.12872) paper. GIoU is a generalized IoU loss for bounding box localization, bbox loss is an L1 bounding box regression loss, and ce is an area-aware sigmoidal focal loss. class error is a percentage accuracy of classification for box classification and cardinality error is not worth paying attention to.
> 
> the rest of the meaning of the loss functions are most directly understood by ensuring familiarity with the code. when we train, we mainly monitor loss_ce_unscaled, class_error_unscaled, loss_bbox_unscaled, and loss_ce_unscaled, as well as their test and ema variants
> 
> does that answer your question?

It's real stuff! I'll check the paper and code carefully.

Many thanks.

**Comment by anyaflya** [comment]
I am also trying to understand the meaning of the loss. What's up regarding the release of the paper? :eyes:

**Comment by isaacrob-roboflow** [comment]
We're still working on it! Prioritizing releasing the best models we can first
